functorch/test/test_eager_transforms.py

```diff
@@ -2014,6 +2014,146 @@ def push_jvp(dummy, x):
         vmap(vmap(push_jvp, (0, None)))(dummy, x)
 
 
+# The tests here follow the cases in [Forward Grad View/inplace]
+# https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/autograd_meta.cpp#L18-L43
+class TestVmapJvpInplaceView(TestCase):
+    # Case 1 in [Forward Grad View/inplace]
+    def test_all_dual_no_view(self, device):
+        B = 2
+
+        def push_jvp(f):
+            def inner(x, xt, y, yt):
+                return jvp(f, (x, y), (xt, yt))
+            return inner
+
+        def f(x, y):
+            x.copy_(y)
+            return x
+        x = torch.randn(3, B, device=device)
+        xt = torch.randn(3, B, device=device)
+        y = torch.randn(3, B, device=device)
+        yt = torch.randn(3, B, device=device)
+        out, out_tangent = vmap(push_jvp(f), in_dims=1)(x, xt, y, yt)
+        self.assertEqual(out, x.movedim(1, 0))
+        self.assertEqual(out_tangent, yt.movedim(1, 0))
+
+        x = torch.randn(3, B, device=device)
+        xt = torch.randn(3, B, device=device)
+        y = torch.randn(3, 3, device=device)[:, 1]
+        yt = torch.randn(6, device=device)[::2]
+        out, out_tangent = vmap(push_jvp(f), in_dims=(1, 1, None, None))(x, xt, y, yt)
+        self.assertEqual(out, x.movedim(1, 0))
+        self.assertEqual(out_tangent, yt.expand(B, 3))
+
+    # Case 2 in [Forward Grad View/inplace]
+    def test_all_dual_base_view_inplace(self, device):
+        B = 2
+
+        def push_jvp(f):
+            def inner(x, xt, y, yt):
+                return jvp(f, (x, y), (xt, yt))
+            return inner
+
+        # with view, propagate from view to base
+        def f(x, y):
+            view = x[:, ::2]
+            view.copy_(y)
+            return view
```

Comment:
nit: for cases 2 and 3 could we have it return both x and view so we can check that both tangents have been properly updated?